### PR TITLE
Use property values along with ConditionedProperties to determine the configurations/platforms to use

### LINKED ISSUE #11: Respect user specified platform and configuration
At the moment, we hard code the Platform and Configuration when generation a solution file.  We should read the values from the project and use those instead.

https://github.com/jeffkl/SlnGen/blob/a25b47386ed6596607d9799ee7632b2f497c956c/src/SlnGen.Build.Tasks/Internal/SlnFile.cs#L60

### PR DESCRIPTION
**Pull request by amrmahdi**
Use property values along with ConditionedProperties to determine the configurations/platforms to use
Fixes #11

### CODE REVIEW

**Comment by jeffkl** [comment]
Looks like a test is failing, does it pass locally?

```
1) Failed : SlnGen.Build.Tasks.UnitTests.SlnProjectTests.ConfigurationsAndPlatforms
  Expected: equivalent to < "amd64", "x64", "AnyCPU" >
  But was:  < "Any CPU", "AnyCPU", "amd64" >
  Missing (1): < "x64" >
  Extra (1): < "Any CPU" >
   at SlnGen.Build.Tasks.UnitTests.SlnProjectTests.ConfigurationsAndPlatforms() in C:\projects\slngen\src\SlnGen.Build.Tasks.UnitTests\SlnProjectTests.cs:line 93
```

**Comment by amrmahdi** [comment]
@jeffkl yes it is passing locally.

**Comment by amrmahdi** [comment]
I'm not sure why is x64 being replaced with Any CPU.

**Comment by jeffkl** [comment]
I added some commits.  It turns out AppVeyor is setting an environment variable `Platform` which was messing up the test.  I added code to clear out `Configuration` and `Platform` environment variables as well as some other minor changes.

File: src/SlnGen.Build.Tasks/ExtensionMethods.cs [hunk 1]
```diff
@@ -106,9 +106,9 @@ public static IEnumerable<string> GetConditionedPropertyValuesOrDefault(this Pro
         /// <returns>The values of the property if one exists, otherwise the default value specified.</returns>
         public static IEnumerable<string> GetPossiblePropertyValuesOrDefault(this Project project, string name, string defaultValue)
         {
-            var propertyValue = project.GetPropertyValue(name);
-            IEnumerable<string> conditionPropertyValues = project.GetConditionedPropertyValuesOrDefault(name, string.Empty).ToList();
-            List<string> values = new List<string>();
+            HashSet<string> values = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```
Comment (amrmahdi) [inline src/SlnGen.Build.Tasks/ExtensionMethods.cs]: Wanted to maintain the actual order, but turned out VS doesn't care and sorts them alphabetically

File: src/SlnGen.Build.Tasks.UnitTests/SlnProjectTests.cs [hunk 2]
```diff
@@ -182,8 +182,16 @@ private sealed class TestProject : IDisposable
...
 
+            private readonly Dictionary<string, string> _savedEnvironmentVariables = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
             private TestProject(string fullPath, IDictionary<string, string> globalProperties)
             {
+                SetEnvironmentVariables(new Dictionary<string, string>
+                {
+                    { "Configuration", null },
+                    { "Platform", null },
+                });
```
Comment (amrmahdi) [inline src/SlnGen.Build.Tasks.UnitTests/SlnProjectTests.cs]: This can't be controlled by AppVeyor configurations ?

Comment (jeffkl) [reply src/SlnGen.Build.Tasks.UnitTests/SlnProjectTests.cs]: Yeah they are set as part of our build.  We need them to be set for other things but not set when these tests run...

**Pull request merged**